newIDE/app/src/EventsSheet/InstructionEditor/TextEditorDialog.js

```diff
@@ -0,0 +1,301 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { t } from '@lingui/macro';
+
+import * as React from 'react';
+import Dialog from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import HelpButton from '../../UI/HelpButton';
+import { Line, Column } from '../../UI/Grid';
+import ColorPicker from '../../UI/ColorField/ColorPicker';
+import MiniToolbar, { MiniToolbarText } from '../../UI/MiniToolbar';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+
+const gd = global.gd;
+
+const toolbarItemStyle = {
+  marginRight: 10,
+};
+
+const styles = {
+  sizeTextField: {
+    width: 90,
+    ...toolbarItemStyle,
+  },
+  toolbarItem: toolbarItemStyle,
+  checkbox: toolbarItemStyle,
+};
+
+type Props = {|
+  event: any,
+  onClose: () => void,
+  onApply: () => void,
+|};
+
+type State = {|
+  textValue: any,
```

Comment:
put proper types there :) This will save us from silly mistakes.

textValue is a `string`
textColors and backgroundColors could be objects like this: `{|r:number, g: number, b: number|}`

(the burger notation `{| |}` indicates that the object can not have other properties than the one listed)

Reply:
I put "any" by default everywhere except when I know the type ;)
I was not sure if an object was a string or other type, thx !